### PR TITLE
Propr propr

### PR DESCRIPTION
**Pull request by suzannejin**
Add propr/propr module to perform compositional data analysis on rna-seq data

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/propr/propr/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    file(params.test_data['mus_musculus']['genome']['rnaseq_matrix'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+        
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.matrix).match("Test propr/propr using default options") },
```
Comment (pinin4fjords) [inline modules/nf-core/propr/propr/tests/main.nf.test:30]: Why only snapshotting the matrix?

File: tests/config/pytest_modules.yml [hunk 2]
```diff
@@ -1,2619 +1,3577 @@
 adapterremovalfixprefix:
   - modules/nf-core/adapterremovalfixprefix/**
   - tests/modules/nf-core/adapterremovalfixprefix/**
+
```
Comment (pinin4fjords) [inline tests/config/pytest_modules.yml:4]: Don't think you need to be changing this file- probably just something funny on your branch.

File: modules/nf-core/propr/propr/tests/main.nf.test (hunk 1 above)
Comment (suzannejin) [reply modules/nf-core/propr/propr/tests/main.nf.test:30]: The FDR file is calculated based on a random sampling, so I don´t think we can snapshot.
Then the rds is the R object including the matrix, and the fdr, and more stuff that could be useful to run propr again if the user wants. So, same problem cannot be snapshot and have the same md5

File: tests/config/pytest_modules.yml (hunk 2 above)
Comment (suzannejin) [reply tests/config/pytest_modules.yml:4]: Cuz it always give me merging conflicts with the master... what should I do then?

File: modules/nf-core/propr/propr/tests/main.nf.test (hunk 1 above)
Comment (pinin4fjords) [reply modules/nf-core/propr/propr/tests/main.nf.test:30]: Can you use set.seed() to make the FDR reproducible? You're right on the rds of course

File: tests/config/pytest_modules.yml (hunk 2 above)
Comment (pinin4fjords) [reply tests/config/pytest_modules.yml:4]: Probably something funny in your git history. Try literally copying in the file from master in a new commit?

File: modules/nf-core/propr/propr/tests/main.nf.test (hunk 1 above)
Comment (suzannejin) [reply modules/nf-core/propr/propr/tests/main.nf.test:30]: That would require to change the package code and rebuild container and so on...
It is really a test that shuffles the input data and run multiple times the function used to create the matrix output, so I think if the matrix output is consistent, this step should not have problem too

Comment (suzannejin) [reply modules/nf-core/propr/propr/tests/main.nf.test]: @pinin4fjords now it should be fine. I added a way to fix the seed for fdr reproducibility

File: modules/nf-core/propr/propr/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                ]
+                """
+            }
+        }
+        
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.matrix).match("Test propr/propr using default options - matrix") },
+                { assert snapshot(process.out.versions).match("versions") }
```
Comment (pinin4fjords) [inline modules/nf-core/propr/propr/tests/main.nf.test:31]: missing fdr check here

File: modules/nf-core/propr/propr/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.matrix).match("Test propr/propr while running clr+pcor.bshrink explicitly - matrix")},
+                { assert snapshot(process.out.fdr).match("Test propr/propr while running clr+pcor.bshrink explicitly - fdr")}
```
Comment (pinin4fjords) [inline modules/nf-core/propr/propr/tests/main.nf.test:56]: Conventionally, we have a version check in every test. But I won't insist.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Thanks for sorting out the seed - I think that really helps. You're just missing the FDR check on the first test.

File: modules/nf-core/propr/propr/tests/main.nf.test (hunk 3 above)
Comment (suzannejin) [reply modules/nf-core/propr/propr/tests/main.nf.test:31]: hey there! fdr is an optional output actually. Here in the default test, I did not ask to compute fdr.

Comment (pinin4fjords) [reply modules/nf-core/propr/propr/tests/main.nf.test:31]: Ahh, my bad, thanks for explanation

**Review by pinin4fjords — APPROVED** [review]

**Comment by suzannejin** [comment]
@pinin4fjords Thank you for the review!

**Pull request merged**